patches/minecraft/net/minecraft/server/level/ServerChunkCache.java.patch

```diff
@@ -11,6 +11,28 @@
                  ChunkAccess chunkaccess1 = chunkholder.getChunkIfPresent(ChunkStatus.FULL);
                  if (chunkaccess1 != null) {
                      this.storeInCache(i, chunkaccess1, ChunkStatus.FULL);
+@@ -368,9 +_,19 @@
+     }
+ 
+     private void collectTickingChunks(List<LevelChunk> p_368970_) {
+-        this.chunkMap.forEachSpawnCandidateChunk(p_358696_ -> {
++        // FORGE: Use a LinkedHashSet to preserve order
++        Set<LevelChunk> set = new java.util.LinkedHashSet<>();
++        this.chunkMap.forEachSpawnCandidateChunk(this.getChunkCollector(set, chunk -> this.level.isNaturalSpawningAllowed(chunk.getPos())));
++        this.chunkMap.forEachForcedChunk(this.getChunkCollector(set, chunk -> true));
```

Comment:
Could just make the second predicate `chunk -> !p_368970_.contains(chunk)` Little costly, but forced chunks should be small enough to not be an issue.

Reply:
It depends. I'm working under the assumption that forced chunks can have any number of chunks in them which is why I delegated to a set in the first place.

Reply:
Good call.